Lib/ssl.py

```diff
@@ -749,8 +749,6 @@ def _create_unverified_context(protocol=None, *, cert_reqs=CERT_NONE,
     context.check_hostname = check_hostname
     if cert_reqs is not None:
         context.verify_mode = cert_reqs
-    if check_hostname:
```

Comment:
We can remove `context.check_hostname = check_hostname` instead, however this would slightly change the current behavior, i.e. `context.check_hostname` will be `None` instead of `False` when `check_hostname` is not passed.